frontend/src/components/Home/index.js

```diff
@@ -2,24 +2,62 @@ import React, { Component } from "react";
 import MainHeader from "./MainHeader";
 import MainContent from "./MainContent"
 import Grid from '@mui/material/Grid';
+import JokeService from "../../services/jokes";
 
 class Home extends Component {
 
-    static initialState = { items: [] };
+    static initialState = { jokes: [], index: 1, loading: false };
 
     constructor(props) {
         super(props);
-        this.state = { items: [] };
+        this.state = { jokes: [], index: 1, loading: false };
+    }
+
+    componentDidMount() {
+        JokeService.getPage(9, this.state.index).then((jokes) => { this.setState({ jokes: jokes }) });
+    }
+
+    _reloadJokes = () => {
```

Comment:
Are you sure these can't be made asynchronous? 
Bit of a shame to have async in the services & do old style callbacks just here

Reply:
They now both use async/await syntax.